packages/odyssey-storybook/src/components/odyssey-mui/Tag/Tag.stories.tsx

```diff
@@ -181,27 +222,33 @@ export const Icon: StoryObj<TagProps> = {
   args: {
     label: "Crew",
     icon: <GroupIcon />,
+    testId: "tag-icon",
   },
 };
 
 export const Clickable: StoryObj<TagProps> = {
   args: {
     label: "Starship",
+    testId: "tag-clickable",
   },
   play: async ({ args, canvasElement, step }) => {
     await step("remove the tag on click", async () => {
-      const canvas = within(canvasElement);
-      const tag = canvas.getByText(args.label);
-      await userEvent.click(tag);
-      expect(args.onClick).toHaveBeenCalledTimes(1);
-      await axeRun("Clickable Tag");
+      const tag = canvasElement.querySelector(`[data-se="${args.testId}"]`);
+      if (tag) {
+        await userEvent.click(tag);
+        expect(args.onClick).toHaveBeenCalledTimes(1);
+        await axeRun("Clickable Tag");
+      } else {
+        throw new Error(`Element with data-se="${args.testId}" not found`);
+      }
```

Comment:
We should use testing library to select these elements.

If we use `getByRole("tab", { name: "" })` instead of `querySelector`, it will throw an error if it can't find the element; therefore, you won't need the `if (tag) {}` check.